In [ ]:
import pandas as pd

df = pd.read_csv('agrofitprodutosformulados.csv', sep=';', engine='pyarrow')
df = df.convert_dtypes()

---


# Dados utilizados na análise

###### Os dados utilizados correspondem a um banco de dados disponibilizado no site do [Agrofit](https://dados.agricultura.gov.br/pt_BR/dataset/sistema-de-agrotoxicos-fitossanitarios-agrofit/resource/d30b30d7-e256-484e-9ab8-cd40974e1238).  
###### Dados acessados no dia 15/01/2026.  
###### Não é possível embutir o link no comando pd.read_csv por conta do erro HTTP Error 403: Forbidden.

In [27]:
df

NR_REGISTRO      MARCA_COMERCIAL  \
0             35523         KBR-829M1-02   
1             35523         KBR-829M1-02   
2             35523         KBR-829M1-02   
3             33723         KBR-S39M1-02   
4             33723         KBR-S39M1-02   
...             ...                  ...   
269996        37019              ÍmparBR   
269997        37019              ÍmparBR   
269998        37019              ÍmparBR   
269999         7697  Óleo Vegetal Nortox   
270000         7697  Óleo Vegetal Nortox   

                                          FORMULACAO  \
0                                   Nematóides vivos   
1                                   Nematóides vivos   
2                                   Nematóides vivos   
3                                   Nematóides vivos   
4                                   Nematóides vivos   
...                                              ...   
269996  FS - Suspensão Concentrada p/ Trat. Sementes   
269997  FS - Suspensão Concentrada p/ Trat. Sementes   
269998  FS - Suspensão Concentrada p/ Trat. Sementes   
269999                 EC - Concentrado Emulsionável   
270000                 EC - Concentrado Emulsionável   

                                        INGREDIENTE_ATIVO  \
0       Heterorhabditis bacteriophora (Nematóides ento...   
1       Heterorhabditis bacteriophora (Nematóides ento...   
2       Heterorhabditis bacteriophora (Nematóides ento...   
3       Steinernema carpocapsae (Nematóides entomopato...   
4       Steinernema carpocapsae (Nematóides entomopato...   
...                                                   ...   
269996              tiametoxam (neonicotinóide) (350 g/L)   
269997              tiametoxam (neonicotinóide) (350 g/L)   
269998              tiametoxam (neonicotinóide) (350 g/L)   
269999  óleo vegetal (ésteres de ácidos graxos) (930 g/L)   
270000  óleo vegetal (ésteres de ácidos graxos) (930 g/L)   

                                    TITULAR_DE_REGISTRO  \
0       Koppert do Brasil Holding S.A. - Piracicaba/SP    
1       Koppert do Brasil Holding S.A. - Piracicaba/SP    
2       Koppert do Brasil Holding S.A. - Piracicaba/SP    
3       Koppert do Brasil Holding S.A. - Piracicaba/SP    
4       Koppert do Brasil Holding S.A. - Piracicaba/SP    
...                                                 ...   
269996                 Ouro Fino Química S.A. - Uberaba   
269997                 Ouro Fino Química S.A. - Uberaba   
269998                 Ouro Fino Química S.A. - Uberaba   
269999                          Nortox S.A. - Arapongas   
270000                          Nortox S.A. - Arapongas   

                              CLASSE MODO_DE_ACAO            CULTURA  \
0       Agente Biológico de Controle         <NA>  Todas as culturas   
1       Agente Biológico de Controle         <NA>  Todas as culturas   
2       Agente Biológico de Controle         <NA>  Todas as culturas   
3       Agente Biológico de Controle         <NA>  Todas as culturas   
4       Agente Biológico de Controle         <NA>  Todas as culturas   
...                              ...          ...                ...   
269996                    Inseticida    Sistêmico              Sorgo   
269997                    Inseticida    Sistêmico              Trigo   
269998                    Inseticida    Sistêmico              Trigo   
269999                    Inseticida  De contato              Citros   
270000                    Inseticida  De contato                <NA>   

           PRAGA_NOME_CIENTIFICO                             PRAGA_NOME_COMUM  \
0           Scaptocoris castanea                           Percevejo-castanho   
1             Sphenophorus levis  Bicudo da cana-de-açúcar;  Gorgulho-da-cana   
2          Spodoptera frugiperda                              Lagarta-militar   
3        Bradysia matogrossensis                                 fungus gnats   
4             Sphenophorus levis  Bicudo da cana-de-açúcar;  Gorgulho-da-cana   
...          

---


## Visão geral do mercado / concentração

### Quantas empresas distintas existem no sistema

In [ ]:
# Criando um dataframe sem duplicatas de produtos com os nomes das empresas detentoras do registro
produtos_df = df[['NR_REGISTRO', 'TITULAR_DE_REGISTRO']].drop_duplicates(subset=['NR_REGISTRO'])

# Retornando o número de registros únicos dos titulares
produtos_df['TITULAR_DE_REGISTRO'].nunique()

333

---
Como podemos ver, são **333** empresas com CNPJ distintos inscritos no banco de dados do Agrofit.

Vale ressaltar que, algumas poucas empresas atuam sob diferentes inscrições.

---

### Top 10 empresas com mais produtos registrados

In [ ]:
# Contagem de quantas vezes os top 10 titulare aparecem no dataframe
produtos_df['TITULAR_DE_REGISTRO'].value_counts().head(10)

TITULAR_DE_REGISTRO
Syngenta Proteção de Cultivos Ltda. – São Paulo/SP                                         248
UPL do Brasil Indústria e Comércio de Insumos Agropecuários S.A. - Matriz Ituverava/SP     193
Adama Brasil S.A.- Londrina/PR                                                             175
Nortox S.A. - Arapongas                                                                    167
Sumitomo Chemical Brasil Indústria Química S.A. - Maracanaú/CE                             165
Rainbow Defensivos Agrícolas Ltda.- Porto Alegre /RS                                       160
CTVA Proteção de Cultivos Ltda - Barueri (Tamboré)                                         144
Basf S.A. – São Paulo                                                                      118
FMC Química do Brasil Ltda. - Campinas/SP                                                  108
Bayer S.A. - São Paulo/ SP                                                                  98
Name: count, dtype: Int64

---

Acima, podemos ver as empresas e a quantidade de produtos registrados por cada uma em ordem decrescente.

Note que, dentre as empresas listada, **apenas 1** delas é genuinamente brasileira: a Nortox S.A., ocupando a quarta colocação com *167* produtos registrados.

---

### Porcentagem de participação das top 10 empresas

In [ ]:
# Porcentagem arredondada do número total de produtos das top 10 empresas comparadas ao total de produtos
round(produtos_df['TITULAR_DE_REGISTRO'].value_counts().head(10).sum() / produtos_df['NR_REGISTRO'].value_counts().sum() * 100, 2)

np.float64(37.2)

In [ ]:
# Contagem de quantos produtos as top 10 empresas têm no cadastro
produtos_df['TITULAR_DE_REGISTRO'].value_counts().head(10).sum()

np.int64(1576)

In [ ]:
# Contagem total de todos os produtos
produtos_df['NR_REGISTRO'].value_counts().sum()

np.int64(4236)

---

Se formos contar a participação que as top 10 empresas têm sob o mercado nacional, elas juntas somam 37,2% do mercado, ou ainda, 1576 produtos dos 4236.

---

### Quantos produtos cada empresa tem por classe das quatro grandes classes (herbicida, fungicida, inseticida e biológico)?

In [ ]:
# Seleciona colunas relevantes e remove duplicatas de produtos
classes_df = df[['NR_REGISTRO', 'TITULAR_DE_REGISTRO', 'CLASSE']].drop_duplicates(subset=['NR_REGISTRO'])

# Cria colunas booleanas para as 4 principais classes
classes_df['HERBICIDA'] = classes_df['CLASSE'].str.contains('Herbicida', case=False, na=False)
classes_df['FUNGICIDA'] = classes_df['CLASSE'].str.contains('Fungicida', case=False, na=False)
classes_df['INSETICIDA'] = classes_df['CLASSE'].str.contains('Inseticida', case=False, na=False)
classes_df['BIOLÓGICO'] = classes_df['CLASSE'].str.contains('biológico', case=False, na=False)

# Agrupa por empresa e soma a quantidade de produtos por classe
classes = classes_df.groupby('TITULAR_DE_REGISTRO')[['HERBICIDA', 'FUNGICIDA', 'INSETICIDA', 'BIOLÓGICO']].sum()

# Adiciona coluna de total para facilitar ordenação
classes['TOTAL'] = classes.sum(axis=1)

# Ordena pelas empresas com mais produtos e exibe as top 10
classes = classes.nlargest(n=10, columns='TOTAL')

classes

HERBICIDA  FUNGICIDA  \
TITULAR_DE_REGISTRO                                                        
Syngenta Proteção de Cultivos Ltda. – São Paulo/SP         48        114   
UPL do Brasil Indústria e Comércio de Insumos A...         86         54   
Adama Brasil S.A.- Londrina/PR                             88         43   
Nortox S.A. - Arapongas                                    84         38   
Rainbow Defensivos Agrícolas Ltda.- Porto Alegr...        114         27   
Sumitomo Chemical Brasil Indústria Química S.A....         71         23   
CTVA Proteção de Cultivos Ltda - Barueri (Tamboré)         77         28   
Basf S.A. – São Paulo                                      30         53   
FMC Química do Brasil Ltda. - Campinas/SP                  31         17   
Bayer S.A. - São Paulo/ SP                                 20         40   

                                                    INSETICIDA  BIOLÓGICO  \
TITULAR_DE_REGISTRO                                                         
Syngenta Proteção de Cultivos Ltda. – São Paulo/SP          84          7   
UPL do Brasil Indústria e Comércio de Insumos A...          41          3   
Adama Brasil S.A.- Londrina/PR                              35          2   
Nortox S.A. - Arapongas                                     42          0   
Rainbow Defensivos Agrícolas Ltda.- Porto Alegr...          17          0   
Sumitomo Chemical Brasil Indústria Química S.A....          54          9   
CTVA Proteção de Cultivos Ltda - Barueri (Tamboré)          30          0   
Basf S.A. – São Paulo                                       29          5   
FMC Química do Brasil Ltda. - Campinas/SP                   50         12   
Bayer S.A. - São Paulo/ SP                                  31          2   

                                                    TOTAL  
TITULAR_DE_REGISTRO                                        
Syngenta Proteção de Cultivos Ltda. – São Paulo/SP    253  
UPL do Brasil Indústria e Comércio de Insumos A...    184  
Adama Brasil S.A.- Londrina/PR                        168  
Nortox S.A. - Arapongas                               164  
Rainbow Defensivos Agrícolas Ltda.- Porto Alegr...    158  
Sumitomo Chemical Brasil Indústria Química S.A....    157  
CTVA Proteção de Cultivos Ltda - Barueri (Tamboré)    135  
Basf S.A. – São Paulo                                 117  
FMC Química do Brasil Ltda. - Campinas/SP             110  
Bayer S.A. - São Paulo/ SP                             93

---
Como podemos ver acima, a partcipação dos grandes players no mercado de biológicos ainda é irrisória frente aos produtos químicos.

Apesar disso, podemos ver a Syngenta ocupando o posto de primeira colocada em número de fungicidas e inseticidas registrados, tendo próximo do dobro de produtos quando comparado às demais empresas.

Quando olhamos para os herbicidas, o mercado é mais competitivo, com 6 de 10 tendo mais 70 produtos registrados, com a chinesa Rainbow encabeçando a lista.

---

## Ingredientes ativos mais usados

### Quais são os 10 ingredientes ativos mais frequentes nos produtos formulados?

In [11]:
ingredientes_df = df[['NR_REGISTRO', 'INGREDIENTE_ATIVO']].drop_duplicates(subset='NR_REGISTRO')
ingredientes_df['INGREDIENTE_ATIVO'] = ingredientes_df['INGREDIENTE_ATIVO'].str.split(' + ')
ingredientes_df = ingredientes_df.explode('INGREDIENTE_ATIVO')

ingredientes_df['INGREDIENTE_ATIVO'].str.strip().value_counts().head(10)

INGREDIENTE_ATIVO
glifosato-sal de isopropilamina (glicina substituída) (480 g/L)    49
Glufosinato - sal de amônio (homoalanina substituída) (200 g/L)    49
cletodim (oxima ciclohexanodiona) (240 g/L)                        36
fipronil (pirazol) (800 g/kg)                                      34
sulfentrazona (triazolona) (500 g/L)                               33
Cotesia flavipes                                                   31
glifosato-sal de amônio (glicina substituída) (792.5 g/kg)         30
clorotalonil (isoftalonitrila) (720 g/L)                           30
clorpirifós (organofosforado) (480 g/L)                            29
acefato (organofosforado) (750 g/kg)                               27
Name: count, dtype: int64

---

Analisando apenas os 10 ingredientes ativos mais utilizados, vemos um domínio do **glifosato**, com 79 produtos contendo alguma formulação do ativo, seguido pelo **Glufosinato**, com 49 produtos utilizando o ativo.

Fipronil segue sendo o primeiro ativo inseticida mais utilizado, mas é seguido por um agente inesperado: ***Cotesia flavipes***.  
Este inseto é uma vespa parasitóide, amplamente utilizada no controle biológico da *broca-da-cana-de-açúcar*.

Dentre os ingredientes ativos, apenas 1 é um fungicida: o **clorotanonil**, ativo multissítio de contato.

---

### Quantos % dos produtos são alguma formulação de Glifosato?

In [12]:
glifosato = ingredientes_df['INGREDIENTE_ATIVO'].str.contains('glifosato', case=False, na=False)
round(ingredientes_df.value_counts(glifosato, normalize=True) * 100, 2)

INGREDIENTE_ATIVO
False    95.65
True      4.35
Name: proportion, dtype: float64

---

Diante da enorme expressividade do **Glifosato**, uma análise curiosa revela que 4,35% dos produtos cadastrados são alguma espécie de Glifosato.

Ou ainda...

In [25]:
ingredientes_df.value_counts(glifosato)

INGREDIENTE_ATIVO
False    4137
True      188
Name: count, dtype: int64

188 produtos de 4236. 

---

### Quais ingredientes ativos podem ser utilizados no maior número de culturas diferentes?

In [13]:
culturas_df = df.groupby('INGREDIENTE_ATIVO')['CULTURA'].nunique()
culturas_df.nlargest(10)

INGREDIENTE_ATIVO
Azoxistrobina (estrobilurina) (200 g/L) + difenoconazol (triazol) (125 g/L)          124
difenoconazol (triazol) (250 g/L)                                                    120
fluxapiroxade (carboxamida) (167 g/L) + piraclostrobina (estrobilurina) (333 g/L)    100
lambda-cialotrina (piretróide) (250 g/L)                                              91
glifosato-sal de isopropilamina (glicina substituída) (480 g/L)                       88
ciprodinil (anilinopirimidina) (750 g/kg)                                             87
diafentiurom (feniltiouréia) (500 g/L)                                                86
Azoxistrobina (estrobilurina) (500 g/kg)                                              85
Swinglea glutinosa (Terpenos), Extrato de (Derivado vegetal) (868.4 g/L)              83
azadiractina (Tetranortriterpenóide) (12 g/L)                                         83
Name: CULTURA, dtype: int64

---

Quando olhamos para 

## Classe de uso

### Distribuição de produtos por classe de uso

In [14]:
distribuicao_df = df[['NR_REGISTRO', 'CLASSE']].drop_duplicates('NR_REGISTRO')
round(distribuicao_df.value_counts(subset='CLASSE', normalize=True).head(10) * 100, 2)

CLASSE
Herbicida                                             33.71
Fungicida                                             17.47
Inseticida                                            15.30
Inseticida Microbiológico                              7.79
Acaricida/Inseticida                                   3.73
Fungicida microbiológico                               3.45
Regulador de Crescimento                               2.57
Nematicida Microbiológico                              2.36
Agente Biológico de Controle                           2.20
Acaricida Microbiológico/Inseticida Microbiológico     1.77
Name: proportion, dtype: float64

In [15]:
distribuicao_df = df[['NR_REGISTRO', 'CLASSE']].drop_duplicates('NR_REGISTRO')
distribuicao_df['CLASSE'] = distribuicao_df['CLASSE'].str.split('/')
distribuicao_explodido_df = distribuicao_df.explode(column='CLASSE')
round(distribuicao_explodido_df.value_counts(subset='CLASSE', normalize=True).head(10) * 100, 2)

CLASSE
Herbicida                       30.05
Inseticida                      19.24
Fungicida                       18.38
Inseticida Microbiológico        8.48
Acaricida                        6.04
Fungicida microbiológico         3.76
Nematicida Microbiológico        2.72
Regulador de Crescimento         2.46
Agente Biológico de Controle     1.94
Acaricida Microbiológico         1.61
Name: proportion, dtype: float64

In [16]:
biologico_df = df[['NR_REGISTRO', 'TITULAR_DE_REGISTRO', 'CLASSE']].drop_duplicates(subset=['NR_REGISTRO'])

biologico_df['BIOLÓGICO'] = biologico_df['CLASSE'].str.contains('biológico', case=False, na=False)

biologico = biologico_df.groupby('TITULAR_DE_REGISTRO')['BIOLÓGICO'].sum()

biologico.nlargest(10)

TITULAR_DE_REGISTRO
Koppert do Brasil Holding S.A. - Piracicaba/SP                             47
Total Biotecnologia Industria e Comercio S/A - Curitiba/PR                 42
Simbiose Biociencias S.A.                                                  34
Bioma Indústria Comércio e Distribuição Ltda - Fazenda Rio Grande/PR       22
Ballagro Agro Tecnologia Ltda. - Bom Jesus dos Perdões/SP                  19
Solubio Tecnologias Agricolas S.A. – Jataí/GO                              18
Vittia S.A. - São Joaquim da Barra 01                                      18
Biota Innovations Industria e Comercio de Bioprodutos Ltda – Uberaba/MG    17
Genica Inovação Biotecnologica S.A.. - Planta 2/Piracicaba/SP              17
Andermatt do Brasil Soluções Biológicas Ltda - ME                          15
Name: BIOLÓGICO, dtype: Int64

## Toxicologia e classe ambiental

### Quantos % dos produtos são classe I (extremamente tóxico), II, III, IV?

In [17]:
df['CLASSE_TOXICOLOGICA'].unique()

<StringArray>
[          'Não Classificado - Produto Não Classificado',
                    'Categoria 4 – Produto Pouco Tóxico',
 'Categoria 5 – Produto Improvável de Causar Dano Agudo',
                                   'Extremamente Tóxico',
                'Categoria 2 – Produto Altamente Tóxico',
            'Categoria 3 – Produto Moderadamente Tóxico',
             'Categoria 1 – Produto Extremamente Tóxico',
                                          'Pouco Tóxico',
          'NÃO DETERMINADO DEVIDO À NATUREZA DO PRODUTO',
                                   'Medianamente Tóxico',
                                      'Altamente Tóxico',
       'BAIXA EXPOSIÇÃO PARA USO RESTRITO EM ARMADILHAS']
Length: 12, dtype: string

In [18]:
mapa_toxico = {
    'Extremamente Tóxico': 'Categoria I / Extremamente Tóxico',
    'Categoria 1 – Produto Extremamente Tóxico': 'Categoria I / Extremamente Tóxico',
    
    'Altamente Tóxico': 'Categoria II / Altamente Tóxico',
    'Categoria 2 – Produto Altamente Tóxico': 'Categoria II / Altamente Tóxico',
    
    'Medianamente Tóxico': 'Categoria III / Moderadamente Tóxico',
    'Categoria 3 – Produto Moderadamente Tóxico': 'Categoria III / Moderadamente Tóxico',
    
    'Pouco Tóxico': 'Categoria IV / Pouco Tóxico',
    'Categoria 4 – Produto Pouco Tóxico': 'Categoria IV / Pouco Tóxico',
    
    'Categoria 5 – Produto Improvável de Causar Dano Agudo': 'Categoria V / Improvável de Causar Dano',
    
    'Não Classificado - Produto Não Classificado': 'Não Classificado',
    'NÃO DETERMINADO DEVIDO À NATUREZA DO PRODUTO': 'Não Determinado',
    'BAIXA EXPOSIÇÃO PARA USO RESTRITO EM ARMADILHAS': 'Uso Restrito (Armadilhas)',
}

df['CLASSE_TOXICOLOGICA_PADRAO'] = df['CLASSE_TOXICOLOGICA'].map(mapa_toxico)

toxicidade_df = df[['NR_REGISTRO', 'CLASSE_TOXICOLOGICA_PADRAO']].drop_duplicates('NR_REGISTRO')
round(toxicidade_df['CLASSE_TOXICOLOGICA_PADRAO'].value_counts(normalize=True) * 100, 2)

CLASSE_TOXICOLOGICA_PADRAO
Categoria V / Improvável de Causar Dano    52.53
Categoria IV / Pouco Tóxico                25.42
Não Classificado                           11.31
Categoria III / Moderadamente Tóxico        5.38
Categoria II / Altamente Tóxico             3.23
Categoria I / Extremamente Tóxico           1.16
Não Determinado                             0.90
Uso Restrito (Armadilhas)                   0.07
Name: proportion, dtype: float64